src/main/java/com/example/demo/controller/ReadFileController.java

```diff
@@ -0,0 +1,24 @@
+package com.example.demo.controller;
+
+import com.example.demo.service.file.CustomFileReader;
+import java.util.List;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```

Comment:
Are you sure this should be a controller, and not internal business logic?

Reply:
after the discussion with Bohdan, I would say that if you want to 100% control the time when you want to load data - then launching this process from controller is ok

but probably you will like the ApplicationListener more
https://reflectoring.io/spring-boot-execute-on-startup/#applicationlistener